### PR TITLE
feat(client): Migrate OfflineWarning to TS

### PR DESCRIPTION
**Pull request by mnlightstone**
Checklist:
- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX
https://github.com/freeCodeCamp/freeCodeCamp/issues/42256

### CODE REVIEW

File: client/src/components/OfflineWarning/offline-warning.tsx [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 import './offline-warning.css';
 
 const delayInMilliSeconds = 5000;
-let id;
-const propTypes = {
-  isOnline: PropTypes.bool.isRequired,
-  isSignedIn: PropTypes.bool.isRequired
-};
-function OfflineWarning({ isOnline, isSignedIn }) {
+let id: NodeJS.Timeout;
```
Comment (ShaunSHamilton) [inline client/src/components/OfflineWarning/offline-warning.tsx:7]:
```suggestion
let id: number;
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Question: I assume as this is not running in Node, we can just use a type of `number`.

File: client/src/components/OfflineWarning/offline-warning.tsx (hunk 1 above)
Comment (mnlightstone) [reply client/src/components/OfflineWarning/offline-warning.tsx]: Ah, this was just what my IDE recommended as a return type.

I get a TS error when using `number`, let me know what you think of my change to `ReturnType<typeof setTimeout>` instead!

Comment (ojeytonwilliams) [reply client/src/components/OfflineWarning/offline-warning.tsx]: I think this is fine. We *could* use window.setTimeout, but then we'd need to check if window existed before using it. This approach avoids that mess, which is a plus.

**Review by ShaunSHamilton — APPROVED** [review]
Thank you, for resolving that. This LGTM 👍

**Pull request merged**